src/Cache.cpp

```diff
@@ -349,18 +349,38 @@ secretName(std::string name, bool internal)
 void
 Cache::loadSecrets(std::vector<std::pair<std::string, bool>> toLoad)
 {
+
+    auto settings = UserSettings::instance()->qsettings();
+
     if (toLoad.empty()) {
         this->databaseReady_ = true;
         emit databaseReady();
+        nhlog::db()->debug("Database ready");
+        return;
+    }
+
+    if (settings->value("run_without_secure_secrets_service").toBool()) {
+        for (auto &[name_, internal] : toLoad) {
+            auto name = secretName(name_, internal);
+            auto value = settings->value("secrets/" + name).toString();
```

Comment:
For profile support, you need to prefix the config values with the profile in some cases. You can check the UserSettings code on how to do that. Otherwise multiple logins will overwrite each other.

Reply:
I tried logging with multiple profiles and it seems ok: the secret don't collide because `secretName` is already generating unique names. This is essentially how the keychain works, I think.

Reply:
Oh, right, I forgot about that.That works then, ty!